components/Forms/RegistrationForm/__tests__/RegistrationForm.test.js

```diff
@@ -52,13 +52,19 @@ describe('RegistrationForm', () => {
     });
 
     const successSpy = jest.fn(() => Promise.resolve(true));
-    const { container, getByText, findByText } = render(
+    const { container, getByText } = render(
       <RegistrationForm onSuccess={successSpy} initialValues={initialValues} />,
     );
 
-    fireEvent.click(getByText('Submit'));
+    const submit = getByText('Submit');
+    fireEvent.click(submit);
+
+    // formik switches to `isSubmitting` in a microtask.
+    // TODO: This should probably synchronously switch to being disabled.
+    await act(async () => {
+      await Promise.resolve();
+    });
 
-    const submit = await findByText('Submit');
```

Comment:
The behavior of the component under test didn't actually changed. The tests only relied on the microtask queue being flushed via `await ...`. The `findByText` was actually irrelevant since the element was already in the DOM (see `getByText('submit')` from earlier.

We only need to flush microtasks to schedule the state update for `isSubmitting`. However, in React 17 those state updates where synchronous so we didn't need to do anything but assert on the outcome. In React 18 **every** state update must be wrapped in `act`. 

So the change can be more minimal with

```js
await act(async () => {
  await findByText('Submit');
});
```
but I opted to make the test more clearer for future reference.